Server/resources/unity_instances.py

```
@@ -20,42 +26,91 @@ async def unity_instances(ctx: Context) -> dict[str, Any]:
-        for inst in instances:
-            name_counts[inst.name] = name_counts.get(inst.name, 0) + 1
+            instances = []
+            for session_id, session_info in sessions.items():
+                project = session_info.get("project") or session_info.get("project_name")
+                project_hash = session_info.get("hash")
 
-        duplicates = [name for name, count in name_counts.items() if count > 1]
+                if not project or not project_hash:
+                    raise ValueError(
+                        "PluginHub session missing required 'project' or 'hash' fields."
+                    )
```

Ruff (0.14.5)

Abstract `raise` to an inner function

(TRY301)